modules/basis/basis-datasource/tomcat-listener/src/main/java/io/dirigible/tomcat/listener/DelegatingObjectFactory.java

```diff
@@ -0,0 +1,141 @@
+package io.dirigible.tomcat.listener;
+
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.NameNotFoundException;
+import javax.naming.NamingException;
+import javax.naming.RefAddr;
+import javax.naming.Reference;
+import javax.naming.spi.ObjectFactory;
+
+public class DelegatingObjectFactory implements ObjectFactory {
```

Comment:
Can't this class be part of the `basic-datasource` project?

Reply:
done